staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,6 +50,194 @@ func TestDecode(t *testing.T) {
+			in:   hex("A2D9EA63614101614102"), // {60003("A"):1, "A":2}
+			into: map[string]interface{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name:  "reject two identical indefinite-length byte string keys split into chunks differently (struct)",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A25F426865436C6C6FFF015F416844656C6C6FFF02"), // {(_ 'he', 'llo'):1, (_ 'h', 'ello'):2}
+			into:  struct{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name: "reject two identical indefinite-length byte string keys split into chunks differently (map)",
+			in:   hex("A25F426865436C6C6FFF015F416844656C6C6FFF02"), // {(_ 'he', 'llo'):1, (_ 'h', 'ello'):2}
+			into: map[string]interface{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name:  "reject two identical indefinite-length text string keys split into chunks differently (struct)",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A27F626865636C6C6FFF017F616864656C6C6FFF02"), // {(_ "he", "llo"):1, (_ "h", "ello"):2}
+			into:  struct{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name:  "reject two identical indefinite-length text string keys split into chunks differently (map)",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A27F626865636C6C6FFF017F616864656C6C6FFF02"), // {(_ "he", "llo"):1, (_ "h", "ello"):2}
+			into:  map[string]interface{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name:  "case-sensitive match",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A2614101614202"), // {"A": 1, "B": 2}
+			into:  s{},
+			want: s{
+				UpperB: 2,
+			},
+			assertOnError: assertNilError,
+		},
+		{
+			name:  "exact match before case-insensitive match",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("a2616101614102"), // {"a": 1, "A": 2}
+			into:  s{},
+			want: s{
+				LowerA: 1,
+			},
+			assertOnError: assertNilError,
+		},
+		{
+			name:  "exact match after case-insensitive match",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A2614101616102"), // {"A": 1, "a": 2}
+			into:  s{},
+			want: s{
+				LowerA: 2,
+			},
+			assertOnError: assertNilError,
+		},
```

Comment:
Please add cases for both these inputs that run in `modes.Decode` and assert that they get an UnknownFieldError (there is an alpha criteria that says we've tested case-insensitive "matches" behave the same as no match at all).